### PR TITLE
Config translations moved to client side (Issue #2093)

### PR DESCRIPTION
**Pull request by matakuka**
probably needs a rebase/ update, as it affects other pending PR

### CODE REVIEW

File: openslides/motions/config_variables.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def get_workflow_choices():
     Returns a list of all workflows to be used as choices for the config variable
     'motions_workflow'. Each list item contains the pk and the display name.
     """
-    return [{'value': str(workflow.pk), 'display_name': ugettext_lazy(workflow.name)}
+    return [{'value': str(workflow.pk), 'display_name': workflow.name}
```
Comment (matakuka) [inline openslides/motions/config_variables.py]: I am unsure of this change, as it is no string to be translated, but a variable pulled from elsewhere.

**Comment by matakuka** [comment]
I am unsure of openslides/motions/config_variables.py:16, as there is a variable instead of a string.

File: openslides/agenda/config_variables.py [hunk 2]
```diff
@@ -3,7 +3,6 @@
 from django.core.exceptions import ValidationError as DjangoValidationError
 from django.core.validators import MaxLengthValidator, MinValueValidator
 from django.utils.translation import ugettext as _
```
Comment (emanuelschuetze) [inline openslides/agenda/config_variables.py]: Please remove also this line and all places where "_" as alias for ugettext is used. Add the strings also to js.

File: openslides/assignments/config_variables.py [hunk 3]
```diff
@@ -18,73 +17,73 @@ def get_config_variables():
...
-        group=ugettext_lazy('Elections'),
-        subgroup=ugettext_lazy('Ballot and ballot papers'))
+        group='Elections',
+        subgroup='Ballot and ballot papers')
 
     # PDF
 
     yield ConfigVariable(
         name='assignments_pdf_title',
         default_value=_('Elections'),
```
Comment (emanuelschuetze) [inline openslides/assignments/config_variables.py]: see above

**Comment by emanuelschuetze** [comment]
one remark, rest looks good.

**Comment by emanuelschuetze** [comment]
> I am unsure of openslides/motions/config_variables.py:16, as there is a variable instead of a string.

Your change is correct because the string of workflow.name is marked in [motions/base.js](https://github.com/OpenSlides/OpenSlides/blob/84eff37bdc6b2451d8450f7edf18faa4b844ff5b/openslides/motions/static/js/motions/base.js#L283) as translateable string.

File: openslides/core/static/js/core/site.js [hunk 4]
```diff
@@ -86,6 +86,116 @@ angular.module('OpenSlidesApp.core.site', [
...
+        gettext('Number of all participants');
+        gettext('Use the following custom number');
+        gettext('Custom number of ballot papers');
+        gettext('Publish election result for elected candidates only (projector view)');
+        gettext('Title for PDF document (all elections)');
+        gettext('Preamble text for PDF document (all elections)');
+        //core
+        gettext('Presentation and assembly system');
+        gettext('Event name');
+        gettext('<a href="http://www.openslides.org">OpenSlides</a> is a free web '
```
Comment (emanuelschuetze) [inline openslides/core/static/js/core/site.js]: For multiline strings in JavaScript you have to use `'...' + '...' ` or `' ...\ ...'`. Then one of the travis errors should be done.

File: openslides/motions/config_variables.py [hunk 5]
```diff
@@ -30,154 +29,154 @@ def get_config_variables():
...
         hidden=True,
         weight=335,
-        group=ugettext_lazy('Motions'),
-        subgroup=ugettext_lazy('Amendments'))
+        group='Motions',
+        subgroup='Amendments')
 
     yield ConfigVariable(
         name='motions_amendments_prefix',
         default_value=pgettext('Prefix for the identifier for amendments', 'A'),
```
Comment (emanuelschuetze) [inline openslides/motions/config_variables.py]: remove the pgettext and just use the string "A" als default_value.
On JavaScript site where you mark 'A' translateable try to add the context 'Prefix for the identifier for amendments' to the gettext function that translators understand what 'A' should be.

**Comment by emanuelschuetze** [comment]
2 remarks

File: openslides/core/static/js/core/site.js [hunk 6]
```diff
@@ -86,6 +86,122 @@ angular.module('OpenSlidesApp.core.site', [
...
+        gettext('Prefix for the identifier for amendments');
+        gettext('Number of (minimum) required supporters for a motion');
+        gettext('Choose 0 to disable the supporting system.');
+        gettext('Supporters');
+        gettext('Remove all supporters of a motion if a submitter edits his' +
+                ' motion in early state');
+        gettext('Title for PDF document (all motions)');
+        gettext('Preamble text for PDF document (all motioqns)');
+        gettext('Show paragraph numbering (only in PDF)');
+        gettext('A','Prefix for the identifier for amendments');
```
Comment (emanuelschuetze) [inline openslides/core/static/js/core/site.js]: Suggestion:
```
/// Prefix for the identifier for amendments
gettext('A');
```
"Insert context comments in JavaScript using the triple forward slash syntax of gettext." (see https://angular-gettext.rocketeer.be/dev-guide/annotate-js/)

**Comment by matakuka** [comment]
Not done yet, working on 'A' in  openslides/core/static/js/core/site.js:179

**Comment by matakuka** [comment]
done

**Comment by emanuelschuetze** [comment]
Looks good now. Ready to merge.

**Comment by normanjaeckel** [comment]
I do not like that the strings of config variables of different apps are defined (all) in the core app. I suggest that we split up the definitions to the base files of the respective apps (motion, agenda, ...).

**Comment by emanuelschuetze** [comment]
@normanjaeckel What do you think about a new i18n.js file per App where we define all translation Strings (for users e.g. also all permissions)? How we can intregrate such i18n.js globally?

**Comment by matakuka** [comment]
I moved all the translations to their app's site.js

**Comment by emanuelschuetze** [comment]
@matakuka Please move the translations to the end of in each site.js file (see e.g. users/site.js). And please add for each config function a short comment ('Mark all permission strings...').

If easy possible please remove the template-en.pot file from your pull request. this file we will update with all translations in a separate commit.

**Comment by matakuka** [comment]
done

File: openslides/assignments/static/js/assignments/site.js [hunk 7]
```diff
@@ -670,6 +670,29 @@ angular.module('OpenSlidesApp.assignments.site', ['OpenSlidesApp.assignments'])
             });
         };
     }
+])
+
+//mark all assignment strings for translation with Javascript
```
Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/site.js]: Use better "... assignment config strings..."

**Comment by emanuelschuetze** [comment]
I checked a last time. Looks good. Ready to merge.
@matakuka Please rebase on top of current master (currently we get a merge conflict because your other pull request was merged before).

**Comment by emanuelschuetze** [comment]
@normanjaeckel Ready to merge.

**Comment by emanuelschuetze** [comment]
@normanjaeckel ping

**Pull request merged**